rules-reviewed/eap6/weblogic/weblogic.windup.xml

```diff
@@ -106,6 +106,22 @@
                     <message>Replace with the Java EE standard method `javax.transaction.TransactionManager.resume(Transaction tx)`.</message>
                     <tag>transactions</tag>
                     <tag>weblogic</tag>
+                    <quickfix name="Use java ee standard method Transaction Manager for resuming 1" type="REPLACE">
+                        <replacement>javax.transaction.TransactionManager.resume</replacement>
+                        <search>weblogic.transaction.TransactionManager.resume</search>
```

Comment:
I thought <search>/<replacement> would specify a string replacement? So I don't see how these would work -- the line the rule triggers on will look something like:

ctm.forceResume(saveTx);

The way these are written implies that the code processing the quickfix would be recognizing a reference to a method call from a particular class, and then replacing it with a call to a method from another class.

The first part (recognizing the reference) we can do, but we would need to have a way to identify whether this quickfix should refer to a string replacement or a reference lookup, right? And the replacement is really not simple at all, because you can't change the type of object the method is being invoked on without finding where that object was created and creating it differently...

I'm guessing for this rule any quickfix would be difficult. Probably we would handle it by offering a quickfix for another rule like the one that triggers on the line where the WebLogic TransactionManager is constructed.